src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionContext.cs

```diff
@@ -182,6 +182,8 @@ public async Task DisposeAsync(bool closeGracefully = false)
         {
             Task disposeTask;
 
+            Cancellation?.Dispose();
```

Comment:
The race still exists if you don’t put it in the lock right? It’s just extremely unlikely.

Reply:
It's very unlikely and would result in a "Failed disposing connection" log.

A new poll would need to be calling cancel while the connection is being disposed.

Reply:
Yes I'm less concerned about that.